lib/ex_unit/lib/ex_unit/assertions.ex

```diff
@@ -529,9 +518,11 @@ defmodule ExUnit.Assertions do
   @max_mailbox_length 10
 
   @doc false
-  def __timeout__(timeout) when is_integer(timeout) and timeout >= 0, do: timeout
+  def __timeout__(timeout, _) when is_integer(timeout) and timeout >= 0, do: timeout
 
-  def __timeout__(timeout),
+  def __timeout__(nil, key), do: Application.fetch_env!(:ex_unit, key)
+
+  def __timeout__(timeout, _),
     do: raise(ArgumentError, "timeout must be a non-negative integer, got: #{inspect(timeout)}")
```

Comment:
Now that we have the key in here, what do you think about including it in the error message?

Reply:
In what way would we use that key? Something like "timeout must be a non-negative integer, got #{timeout} - set this default in ex_unit.#{key}"?

I'm not totally sure that would be helpful for folks, but I'm happy to include it if y'all think it would be a good thing.

Reply:
Yeah, I don't see how we can use the key here given that we use it only when a timeout is not given?

Reply:
I thought we could use it for saying something like:

> assert_receive timeout must be a non-negative integer, got: ...

But nevermind, probably not worth it